src/ripple/app/paths/impl/StrandFlow.h

```diff
@@ -468,7 +479,8 @@ flow (PaymentSandbox const& baseView,
 
             activeStrands.push (strand);
 
-            if (!best || q > best->quality)
+            if (!best || best->quality < q ||
+                (best->quality == q && best->out < f.out))
                 best.emplace (f.in, f.out, std::move (*f.sandbox), *strand, q);
         }
 
```

Comment:
While we're in this code: the `bool const shouldBreak = !bool(best)` immediately below could simply become:

    if (best)